src/main/kotlin/org/hyacinthbots/lilybot/extensions/events/AutoThreading.kt

```diff
@@ -0,0 +1,529 @@
+							).contains(event.message.type) ||
+							listOf(
+								ChannelType.GuildNews,
+								ChannelType.GuildVoice,
+								ChannelType.PublicGuildThread,
+								ChannelType.PublicNewsThread
+							).contains(event.message.getChannelOrNull()?.type)
+				}
+			}
+			action {
+				onMessageSend(event, event.getMessageOrNull(), event.pkMessage)
+			}
+		}
+
+		event<UnProxiedMessageCreateEvent> {
+			check {
+				anyGuild()
+				failIf {
+					event.message.author?.id == kord.selfId ||
+							listOf(
+								MessageType.ChatInputCommand,
+								MessageType.ThreadCreated,
+								MessageType.ThreadStarterMessage
+							).contains(event.message.type) ||
+							listOf(
+								ChannelType.GuildNews,
+								ChannelType.GuildVoice,
+								ChannelType.PublicGuildThread,
+								ChannelType.PublicNewsThread
+							).contains(event.message.getChannelOrNull()?.type)
+				}
+			}
+			action {
+				onMessageSend(event, event.getMessageOrNull())
+			}
+		}
+
+		event<ThreadChannelCreateEvent> {
+			check {
+				anyGuild()
+				failIf {
+					event.channel.ownerId == kord.selfId ||
+							event.channel.member != null
+				}
+			}
+			action {
+				// fixme this event fires twice for some unknown reason so this is a workaround
+				delay(1000)
+				if (event.channel.getLastMessage()?.withStrategy(EntitySupplyStrategy.rest) != null) return@action
+
+				val thread = event.channel.asChannelOfOrNull<TextChannelThread>() ?: return@action
+				val options = AutoThreadingCollection().getSingleAutoThread(thread.parentId) ?: return@action
+
+				handleThreadCreation(
+					options,
+					thread,
+					event.channel.owner.asUser()
+				)
+			}
+		}
+	}
+
+	inner class AutoThreadingArgs : Arguments() {
+		val role by optionalRole {
+			name = "role"
+			description = "The role, if any, to invite to threads created in this channel."
+		}
+
+		val preventDuplicates by defaultingBoolean {
+			name = "prevent-duplicates"
+			description = "If users should be stopped from having multiple open threads in this channel. Default false."
+			defaultValue = false
+		}
+
+		val archive by defaultingBoolean {
+			name = "archive"
+			description = "If threads should be archived on creation to avoid filling the sidebar. Default false."
+			defaultValue = false
+		}
+
+		val contentAwareNaming by defaultingBoolean {
+			name = "content-aware-naming"
+			description = "If Lily should use content-aware thread titles. Default false"
+			defaultValue = false
+		}
+
+		val mention by defaultingBoolean {
+			name = "mention"
+			description = "If the creator should be mentioned in new threads in this channel. Default false."
+			defaultValue = false
+		}
+
+		val message by defaultingBoolean {
+			name = "message"
+			description = "Whether to send a custom message on thread creation or not. Default false."
+			defaultValue = false
+		}
+	}
+
+	private suspend fun createMessageModal(inputInteraction: ModalParentInteractionBehavior): String? {
```

Comment:
This can probably be converted to the new KordEx modals. I can provide support if necessary, but this is currently out of line with how the rest of the bot is

Reply:
We don't always send it though, so I don't think that's possible.

Reply:
It is infact possible, if you look into the support and mod config systems you can find the changed modal setup

Reply:
Done